x-pack/plugins/synthetics/server/synthetics_service/project_monitor_formatter.ts

```diff
@@ -228,15 +260,78 @@ export class ProjectMonitorFormatter {
     return savedObjects?.[0];
   };
 
-  private createMonitor = async (normalizedMonitor: BrowserFields) => {
-    await syncNewMonitor({
-      normalizedMonitor,
-      monitor: normalizedMonitor,
-      server: this.server,
-      syntheticsMonitorClient: this.syntheticsMonitorClient,
-      savedObjectsClient: this.savedObjectsClient,
-      request: this.request,
-    });
+  private createMonitorsBulk = async (monitors: BrowserFields[]) => {
+    try {
+      if (monitors.length > 0) {
+        const { newMonitors } = await syncNewMonitorBulk({
+          normalizedMonitors: monitors,
+          server: this.server,
+          syntheticsMonitorClient: this.syntheticsMonitorClient,
+          soClient: this.savedObjectsClient,
+          request: this.request,
+          privateLocations: this.privateLocations,
+          spaceId: this.spaceId,
+        });
+
+        if (newMonitors && newMonitors.length === monitors.length) {
+          this.createdMonitors.push(...monitors.map((monitor) => monitor[ConfigKey.JOURNEY_ID]!));
+          this.handleStreamingMessage({
+            message: `${monitors.length} monitor${
+              monitors.length > 1 ? 's' : ''
+            } created successfully.`,
+          });
+        } else {
+          this.failedMonitors.push({
+            id: `${monitors.length}`,
+            reason: `Failed to create ${monitors.length} monitors`,
+            details: 'Failed to create monitors',
+            payload: monitors,
+          });
+          this.handleStreamingMessage({
+            message: `Failed to create ${monitors.length} monitors`,
+          });
+        }
+      }
+    } catch (e) {
+      this.server.logger.error(e);
+      this.failedMonitors.push({
+        id: `${monitors.length}`,
+        reason: `Failed to create ${monitors.length} monitors`,
+        details: e.message,
+        payload: monitors,
+      });
+      this.handleStreamingMessage({
+        message: `Failed to create ${monitors.length} monitors`,
+      });
+    }
+  };
+
+  private updateMonitorBulk = async (monitors: BrowserFields[]) => {
+    try {
+      for (const monitor of monitors) {
+        const previousMonitor = await this.getExistingMonitor(monitor[ConfigKey.JOURNEY_ID]!);
+        await this.updateMonitor(previousMonitor, monitor);
```

Comment:
Ideally, we would catch errors for each individual update request, since we're not yet doing bulk operations on update and there could be errors that apply to an individual monitor, rather than a group of monitors.

Since we are moving forward bulk operations for update, I can overlook this now, but if we don't get the bulk update PR merged by FF, I will raise a bug.